src/modules/0.0.21/aws_memory_db/entity/memory_db_cluster.ts

```diff
@@ -56,13 +56,20 @@ export class MemoryDBCluster {
   })
   securityGroups?: SecurityGroup[];
 
-  @ManyToOne(() => SubnetGroup, subnetGroup => subnetGroup.subnetGroupName, {
+  @ManyToOne(() => SubnetGroup, {
     nullable: false,
     eager: true,
   })
-  @JoinColumn({
-    name: 'subnet_group',
-  })
+  @JoinColumn([
+    {
+      name: 'subnet_group',
+      referencedColumnName: 'subnetGroupName',
+    },
+    {
+      name: 'region',
+      referencedColumnName: 'region',
+    },
```

Comment:
Does this automatically create a `region` column under the hood, or do you need to also define a `region` column in the class? If it's the former, *how* do you get access to the `region` here? Always through the `subnetGroup`? And if that's the answer, what do we do if the join is nullable?

Reply:
All assumptions are correct and it will fail or not work for nullable joins. It will also fail in the case where you have multiple joins because it will be hard to follow which one is the source of truth. 

Maybe a better solution is always to have the `region` defined as an FK to `aws_regions`, and additionally, we add a manual constraint that forces the region equality between all the related n entities.

Reply:
What does `TypeORM` generate in the migration file if you just copy-paste the `region: string` definition like you have in the `SubnetGroup` class along with this definition? Perhaps it creates both FK relations?